File: hashtable/136/136.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        num_set=set()
        for n in nums:
            if n in num_set:
                num_set.remove(n)
            else:
                num_set.add(n)
        return list(num_set)[0]

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        a=0
        for n in nums:
            a^=n
        return a

File: hashtable/136/136.js
// solution 1
var singleNumber = function(nums) {
  let set = new Set();
  for (n of nums) {
    if (!set.has(n)) {
      set.add(n);
    } else {
      set.delete(n);
    }
  }
  return set.values().next().value;
};

//solution 2
var singleNumber = function(nums) {
  let result = 0;
  for (n of nums) {
    result ^= n;
  }
  return result;
};
